tests/NodeManipulator/ClassDependencyManipulatorTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Tests\NodeManipulator;
+
+use PhpParser\Node\Const_;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassConst;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\Stmt\PropertyProperty;
+use PhpParser\PrettyPrinter\Standard;
+use PHPStan\Type\ObjectType;
+use Rector\NodeManipulator\ClassDependencyManipulator;
+use Rector\PostRector\ValueObject\PropertyMetadata;
+use Rector\Testing\PHPUnit\AbstractLazyTestCase;
+
+final class ClassDependencyManipulatorTest extends AbstractLazyTestCase
+{
+    private ClassDependencyManipulator $classDependencyManipulator;
+
+    private Standard $printerStandard;
+
+    protected function setUp(): void
+    {
+        $this->classDependencyManipulator = $this->make(ClassDependencyManipulator::class);
+        $this->printerStandard = new Standard();
+    }
+
+    public function testEmptyClass(): void
+    {
+        $someClass = new Class_(new Identifier('EmptyClass'));
+
+        $this->addSingleDependency($someClass);
+        $this->asssertClassEqualsFile($someClass, __DIR__ . '/Fixture/expected_empty_class.php.inc');
+    }
+
+    public function testSingleMethod(): void
+    {
+        $someClass = new Class_(new Identifier('SingleMethodClass'));
+        $someClass->stmts[] = new ClassMethod('firstMethod');
+
+        $this->addSingleDependency($someClass);
+
```

Comment:
str_replace `\n` with PHP_EOL can be added first to handle windows, see example

https://github.com/rectorphp/rector-src/blob/1b1807b0f8426641e13e5914389e1c118b3a3d14/tests/PhpParser/Printer/BetterStandardPrinterTest.php#L41

Reply:
:+1:

Reply:
:+1: